### ISSUE TITLE
Deleting a guiWire instance causes crash

### ISSUE THREAD

**Issue by jakeprem** [CedarvilleCS/CedarLogic#1]
To recreate the bug: 
1. Start with or without debugging.
2. Open a file.
- Opening a file while on a blank canvas works.
  1. Open another file. This will cause the program to crash

In file `GUICircuit.cpp`, lines `37-41`.

`thisWire` is a pair, containing two objects. `first` is an unsigned long, and `second` is an object of type guiWire. When opening a new file CedarLogic first deletes all wires (contained in `wireList`) and all gates  (contained in `gateList`). Currently it crashes while iterating through the wires in wirelist, specifically at line `39`. This is when `delete thisWire->second;` is called. Something is happening to uninitialize the wires before the delete is caused.

This is pretty similar to what happened with the guiGate destructor. In that case, the hotspot list contained pairs with pointers to certain klsCollisionObjects that are also still contained as subobjects of the main klsCollisionObject. guiGate destructor deleted the objects, leaving the subObjs list with a set of uninitialized objects to be deleted.

Basically in this case I think it's probable that there are two pointers pointing to the wires, and the wires are getting deleted somewhere else before reaching this crashing code. I recommend walking the call stack and trying to map out which objects are pointing to what and where they're coming from.

![image](https://cloud.githubusercontent.com/assets/10181938/15059220/14fc9f38-12ef-11e6-882b-ec49d887d02b.png)
If you right click on the `[External Code]` in the call stack you can expand it.
